### PR TITLE
Mark IQueryable.DefaultIfEmpty() as nullable

### LINKED ISSUE #69817: IQueryable.DefaultIfEmpty() marked as non-nullable even though it can return null values as a LEFT JOIN
### Description

`IQueryable.DefaultIfEmpty()`, unlike it's normal Linq equivalent, returns `IQueryable<TSource>` instead of `IQueryable<TSource?>`.

### Reproduction Steps

Simple enough, here is a simple repro which should warn for a null reference but instead doesn't:
```cs
var users = applicationDbContext.Users.Where(x => false).DefaultIfEmpty().ToList();
foreach (var user in users ) Console.WriteLine(user.UserName); // NullReferenceException unmarked by the compiler
```

### Expected behavior

`DefaultIfEmpty()` should have the return type of `IQueryable<TSource?>`.

### Actual behavior

See above. The behaviour is especially messy because the non-nullability prompts the IDE to remove null checks on the supposedly-not-null property.

### Regression?

_No response_

### Known Workarounds

Project to a new object and cast to nullable manually:
```cs
applicationDbContext.MyTable.Where(x => false).DefaultIfEmpty().Select(x => (TSource?)x)...
```
not ideal, though.

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by kanadaj**
Fixes https://github.com/dotnet/runtime/issues/69817, very small change overall

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=69821) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=69821) before we can accept your contribution.<br/><br/>:x: kanadaj [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=69821)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=69821) it.</sub>

File: src/libraries/System.Linq.Queryable/src/System/Linq/Queryable.cs [hunk 1]
```diff
@@ -1311,7 +1311,7 @@ public static TSource ElementAt<TSource>(this IQueryable<TSource> source, Index
         }
 
         [DynamicDependency("DefaultIfEmpty`1", typeof(Enumerable))]
-        public static IQueryable<TSource> DefaultIfEmpty<TSource>(this IQueryable<TSource> source)
+        public static IQueryable<TSource?> DefaultIfEmpty<TSource>(this IQueryable<TSource> source)
```
Comment (eiriktsarpalis) [inline src/libraries/System.Linq.Queryable/src/System/Linq/Queryable.cs:1314]: Please update the signature in the reference source as well:

https://github.com/dotnet/runtime/blob/5d3288d449ed61fa1e9da3fe876400ac14c47fa7/src/libraries/System.Linq.Queryable/ref/System.Linq.Queryable.cs#L78

Comment (kanadaj) [reply src/libraries/System.Linq.Queryable/src/System/Linq/Queryable.cs:1314]: Updated in https://github.com/kanadaj/runtime/commit/9e9e6dae6cd3dc3605ed199fd9b9f879bd7a0326, thanks for the heads-up!

**Review by eiriktsarpalis — APPROVED** [review]
Thanks!

**Pull request merged**